index.js

```diff
@@ -11,20 +11,50 @@ const defaults = {
 };
 
 function appleCheck(options) {
-	return got('http://captive.apple.com/hotspot-detect.html', {
+	const gotPromise = got('http://captive.apple.com/hotspot-detect.html', {
 		family: options.version === 'v4' ? 4 : 6,
 		headers: {'User-Agent': 'CaptiveNetworkSupport/1.0 wispr'}
-	}).then(res => /Success/.test(res.body || '') || Promise.reject());
+	});
+
+	const promise = gotPromise.then(res => {
+		return /Success/.test(res.body || '') || Promise.reject();
+	}).catch(err => {
+		if (!(err instanceof got.CancelError)) {
+			throw err;
+		}
+	});
+
+	promise.cancel = gotPromise.cancel;
+
+	return promise;
```

Comment:
Maybe there's a more elegant way of attaching the `.cancel` handler to the `.then` expression.